rslabel_package/rslabel/ret.py

```diff
@@ -97,12 +101,49 @@ def app(args):
                     break
             src_dir = 'in_progress/validation/'
             dest_dir = 'done/' if complete else 'unvalidated/'
+        elif tar_name in clarification_tars:
+            delete = True
+            complete = True
+
+            dir_name = os.path.dirname(args.annotations)
+
+            # Looping over to see if there are bad images, and see if the tar is complete
+            for annotation in json_contents:
+                try:
+                    ann = annotation['status']
+                    if ann == 'Bad':
+                        os.remove("{}/{}".format(dir_name, annotation['filename']))
+                except:
+                    complete = False
+                    pass
+            # If we need to delete json file, because it has bad images
```

Comment:
Don't delete the JSON file - just update the JSON in the loop above and then overwrite the previous json with a new one. It simplifies the logic and reduces the chance for bugs.

Reply:
This is being deleted if we delete few images, there is not need to modify JSON, as creating new for each remaining images in folder would reduce bugs of modifications. It is a tradeoff, any of them can have bugs.